### ISSUE TITLE
New instances were placed on an incompletely added sled

### ISSUE THREAD

**Issue by askfongjojo** [oxidecomputer/omicron#5530]
A newly added sled which didn't succeed in getting its NTP zone created (#5502) was still in use by nexus for new instance placement. Its state was set to `active` in the `sled` table once it has gone through the `omdb --destructive nexus sleds add` step.

According to [RFD 457](https://rfd.shared.oxide.computer/rfd/0457#_sled_policy_stored_in_the_sled_table), the process should involve a state transition from `initializing` to `in-service` to prevent using a sled that is partially initialized for customer workload or other service zones.

**Comment by askfongjojo** [comment]
That's correct. Disks didn't land on the sled because the dataset records haven't been inserted into the CRDB table which disk placement query goes against.

**Comment by morlandi7** [comment]
Something similar was seen on R17 dogfood update. This issue will be updated to more generally track a fix for preventing VMs from being placed on sleds that are still being time-synced
